lib/queue_classic_plus/tasks/work.rake

```diff
@@ -16,19 +16,17 @@ namespace :qc_plus do
     trap('INT') do
       $stderr.puts("Received INT. Shutting down.")
       if !@worker.running
-        $stderr.puts("Worker has stopped running. Exit.")
-        exit(1)
+        $stderr.puts("Worker has already stopped running.")
       end
       @worker.stop
-      exit
     end
 
     trap('TERM') do
       $stderr.puts("Received Term. Shutting down.")
       @worker.stop
-      exit
     end
 
     @worker.start
+    $stderr.puts 'Shut down successfully'
```

Comment:
The QC worker API is a bit confusing: `start` actually blocks "forever" (should really be called `run` I guess) until `@running` is `false`. `stop` sets `@running` to `false`. I really have no idea how it's "supposed" to work (since `start` is blocking and `stop` isn't and also isn't thread-safe), but the `trap` block seems to do the right thing (i.e. set `@running` to `false` so that the worker won't pick up any new jobs and will exit). So this `puts` only gets called after everything's finished.